src/Interpreters/Cache/LRUFileCachePriority.cpp

```diff
@@ -209,15 +209,26 @@ void LRUFileCachePriority::iterate(IterateFunc && func, const CachePriorityGuard
             continue;
         }
 
-        if (metadata->size() != entry.size)
+        if (metadata->size(FileSegment::SizeAlignment::NOT_ALIGNED) != entry.getSize(IFileCachePriority::Entry::SizeAlignment::NOT_ALIGNED))
         {
             throw Exception(
                 ErrorCodes::LOGICAL_ERROR,
                 "Mismatch of file segment size in file segment metadata "
                 "and priority queue: {} != {} ({})",
-                entry.size.load(), metadata->size(), metadata->file_segment->getInfoForLog());
+                entry.getSize(IFileCachePriority::Entry::SizeAlignment::NOT_ALIGNED), metadata->size(FileSegment::SizeAlignment::NOT_ALIGNED), metadata->file_segment->getInfoForLog());
         }
 
+        if (metadata->size() != entry.getSize())
+        {
+            throw Exception(
+                ErrorCodes::LOGICAL_ERROR,
+                "Mismatch of entry file segment size in file segment metadata "
```

Comment:
```suggestion
                "Mismatch of file segment size in file segment metadata "
```
(As it used to be before)